packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1378,10 +1434,32 @@ class _TabBarState extends State<TabBar> {
     return true;
   }
 
+  bool _debugTabAlignmentIsValid(TabAlignment tabAlignment) {
+    assert(() {
+      if (widget.isScrollable && tabAlignment == TabAlignment.fill) {
+        throw FlutterError(
+          '$tabAlignment is only valid for non-scrollable tab bars.',
+        );
+      }
+      if (!widget.isScrollable
```

Comment:
Reading through these cases now, it is a bit clearer which is allowed. Although, I am curious why a scrollable one cannot be aligned at the start?

It is ok to expose options that deviate from the spec if that is why.

Reply:
> Reading through these cases now, it is a bit clearer which is allowed. Although, I am curious why a scrollable one cannot be aligned at the start?

They can be aligned. if `scrollable` is true and tab alignment is `TabAlignment.start`, it won't throw an error.